source/common/http/websocket/ws_handler_impl.cc

```diff
@@ -46,6 +48,11 @@ Network::FilterStatus WsHandlerImpl::onData(Buffer::Instance& data, bool end_str
   ENVOY_LOG(debug, "WsHandlerImpl::onData with buffer length {}, end_stream == {}", data.length(),
             end_stream);
 
+  if (init_failure_) {
```

Comment:
it would be nice if this logic can be moved into the if-else loop below, such that is_connected_ can become some sort of an ENUM - connecting, connected, failed_to_connect, that can be handled by a switch case.